xhr.js

```diff
@@ -0,0 +1,30 @@
+var APP = APP || {};
+(function(app) {
+
+    var API_URLROOT = 'https://api.stackexchange.com/2.2/questions/';
+
+    app.xhr = {};
+
+    app.xhr.fetch = function(ids) {
+        return new Promise(function(resolve, reject) {
+            var req = new XMLHttpRequest();
+            var url = app.xhr.makeApiUrl(ids);
+            req.onreadystatechange = function() {
+                if (req.readyState === 4) {
+                    if (req.status === 200) {
+                        return resolve(JSON.parse(req.responseText));
+                    } else {
+                        return reject('shit dun goofed');
```

Comment:
But if I do https://github.com/JavaScriptRoom/JavaScriptRoom.github.io/pull/1#discussion_r29100340? and get a JSON responseType I can get rid of the entire JSON.parse call. Question is, what happens then if the SO API returns invalid JSON? I would guess I don't get a 200 status in the first place?